goal_src/jak2/engine/mods/card-display.gc

```diff
@@ -0,0 +1,456 @@
+    )
+    (when (logtest? but (pad-buttons l2))
+      (cat-string<-string (-> this temp-but-string) "<PAD_L2>")
+    )
+    (when (logtest? but (pad-buttons r2))
+      (cat-string<-string (-> this temp-but-string) "<PAD_R2>")
+    )
+    (when (logtest? but (pad-buttons l1))
+      (cat-string<-string (-> this temp-but-string) "<PAD_L1>")
+    )
+    (when (logtest? but (pad-buttons r1))
+      (cat-string<-string (-> this temp-but-string) "<PAD_R1>")
+    )
+    (when (logtest? but (pad-buttons triangle))
+      (cat-string<-string (-> this temp-but-string) "<PAD_TRIANGLE>")
+    )
+    (when (logtest? but (pad-buttons circle))
+      (cat-string<-string (-> this temp-but-string) "<PAD_CIRCLE>")
+    )
+    (when (logtest? but (pad-buttons x))
+      (cat-string<-string (-> this temp-but-string) "<PAD_X>")
+    )
+    (when (logtest? but (pad-buttons square))
+      (cat-string<-string (-> this temp-but-string) "<PAD_SQUARE>")
+    )
+  (pc-encode-utf8-string (-> this temp-but-string) (-> this temp-encoded-string))
+  (-> this temp-encoded-string))
+
+
+;;;;;;;;;;;;;;
+;;Menu code
+;;;;;;;;;;;;;;
+
+(defun card-preset-setter ((stick symbol) (click symbol) (x int) (y int) (color rgba) (m int))
+  (set! *show-stick?* stick)
+  (set! *ignore-click?* click)
+  (set! *card-x-pos* x)
+  (set! *card-y-pos* y)
+  (set! *card-r* (the int (-> color r)))
+  (set! *card-g* (the int (-> color g)))
+  (set! *card-b* (the int (-> color b)))
+  (set! *card-a* (the int (-> color a)))
+  (set! MAX_CARDS m)
+  (card-display-off)
+  (card-display-on)
+  (none)
+)
+
+;;a copy of dm-int-var-func, added an off/on to rebuild array when MAX_CARDS is changed
+(defun dm-int-var-func-cards ((var symbol) (msg debug-menu-msg) (val int) (undo-val int))
+  (when (= msg (debug-menu-msg press))
+    (set! (-> var value) val)
+
+    (card-display-off)
+    (card-display-on)
+    )
+  (-> var value)
+  )
+
+
+;;this adds the Input Cards menu to the debug menu
+(when *debug-segment*
+  (when (-> *debug-menu-context* root-menu)
+    (debug-menu-append-item (-> *debug-menu-context* root-menu)
+      (debug-menu-make-from-template *debug-menu-context*
+        '(menu "Input Cards"
+            (function "On/Off" #f ,(lambda () (if (not *input-cards-on?*) (card-display-on) (card-display-off))))
+            (menu "Presets"
+              (function "Default" #f ,(lambda () (card-preset-setter #t #f 25 10 (static-rgba 0 0 128 40) 20)))
+              (function "Right" #f ,(lambda () (card-preset-setter #t #f 455 0 (static-rgba 0 0 0 127) 19)))
+              (function "Corner" #f ,(lambda () (card-preset-setter #t #f -5 395 (static-rgba 240 100 200 70) 1)))
+              (function "Action" #f ,(lambda () (card-preset-setter #f #t 280 250 (static-rgba 110 110 110 15) 6)))
+              (function "British" *british* dm-boolean-toggle-pick-func)
```

Comment:
its a bit weird that this is a boolean toggle under this menu (it doesnt show highlight properly when on/off as a result), maybe just move it up one level in the menus?